source/ambassador_host_test.go

```diff
@@ -141,3 +658,77 @@ func TestParseAmbLoadBalancerService(t *testing.T) {
 		}
 	}
 }
+
+func convertAmbassadorHostToUnstructured(hp *ambassador.Host, s *runtime.Scheme) (*unstructured.Unstructured, error) {
+	unstructuredAmbassadorHost := &unstructured.Unstructured{}
+	if err := s.Convert(hp, unstructuredAmbassadorHost, context.Background()); err != nil {
+		return nil, err
+	}
+	return unstructuredAmbassadorHost, nil
+}
+
+func newAmbassadorDynamicKubernetesClient() (*fakeDynamic.FakeDynamicClient, *runtime.Scheme) {
+	s := runtime.NewScheme()
+	_ = ambassador.AddToScheme(s)
+	return fakeDynamic.NewSimpleDynamicClient(s), s
+}
+
+type fakeAmbassadorHost struct {
+	namespace   string
+	name        string
+	annotations map[string]string
+	hostname    string
+	labels      map[string]string
+}
+
+func (ir fakeAmbassadorHost) Host() *ambassador.Host {
+	spec := ambassador.HostSpec{
+		Hostname: ir.hostname,
+	}
+
+	host := &ambassador.Host{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   ir.namespace,
+			Name:        ir.name,
+			Annotations: ir.annotations,
+			Labels:      ir.labels,
+		},
+		Spec: &spec,
+	}
+
+	return host
+}
+
+type fakeAmbassadorLoadBalancerService struct {
```

Comment:
I don't see anything ambassador-specific in this struct. Can't one reuse https://github.com/kubernetes-sigs/external-dns/blob/0bf5faed42215ce20533de385bb6dccde10c4ab1/source/contour_httpproxy_test.go#L52 ?

Possibly move this to e.g https://github.com/kubernetes-sigs/external-dns/blob/0bf5faed42215ce20533de385bb6dccde10c4ab1/source/source_test.go